File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/i18n_resource.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import PlusApiResultListLabelOption

class I18nResourceApi:
    """i-18n-resource-controller API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def get_options(self, params: Optional[Dict[str, Any]] = None) -> PlusApiResultListLabelOption:
        return self._client.get(f"/ai/v3/v1/open/i18n/options", params=params)

    def create_options(self, params: Optional[Dict[str, Any]] = None) -> PlusApiResultListLabelOption:
        return self._client.post(f"/ai/v3/v1/open/i18n/options", params=params)

File: sdkwork-ai-sdk-go/api/i18n_resource.go
package api

import (
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type I18nResourceApi struct {
    client *sdkhttp.Client
}

func NewI18nResourceApi(client *sdkhttp.Client) *I18nResourceApi {
    return &I18nResourceApi{client: client}
}

func (a *I18nResourceApi) GetOptions(query map[string]interface{}) (sdktypes.PlusApiResultListLabelOption, error) {
    raw, err := a.client.Get(AiApiPath("/v1/open/i18n/options"), query, nil)
    if err != nil {
        var zero sdktypes.PlusApiResultListLabelOption
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListLabelOption](raw)
}

func (a *I18nResourceApi) CreateOptions(query map[string]interface{}) (sdktypes.PlusApiResultListLabelOption, error) {
    raw, err := a.client.Post(AiApiPath("/v1/open/i18n/options"), nil, query, nil, "")
    if err != nil {
        var zero sdktypes.PlusApiResultListLabelOption
        return zero, err
    }
    return decodeResult[sdktypes.PlusApiResultListLabelOption](raw)
}
